source/material_model/visco_plastic.cc

```diff
@@ -39,6 +39,11 @@ namespace aspect
       for ( unsigned int i=0; i < x_comp.size(); ++i)
         x_comp[i] = std::min(std::max(x_comp[i], 0.0), 1.0);
 
+      // assign compositional fields associated with strain a value of 0
+      if (use_strain_weakening == true)
```

Comment:
This is a tricky issue.  If the users wish to use to the `integrated_strain` compositional field plugin, then the strain fields do need to be listed first and in the exact order specified in the documentation.  I agree that using the scheme you outlined above is a better method, but this seems to require some restructuring of the integrated_strain plugin and potentially other compositional field plugins.  Is this a broader topic that should be addressed?